css-fonts-5/Overview.bs

```diff
@@ -59,7 +60,78 @@ rules defined in CSS Fonts Level 4.
+value of <code>legacy</code>.
+
+<h3 id="text-scale-meta-keywords">
+Keywords</h3>
+
+The recognized keywords in the [=text-scale=]
+<{meta}> element are:
+
+<dl>
+	<dt><dfn for="text-scale"><code>legacy</code></dfn></dt>
+	<dd>''env()/preferred-text-scale'' returns the user's OS-level font factor on mobile,
+	but 1 on desktop.
+	The document's initial font size only incorporates UA-level font preferences.
+	OS-level font preferences are ignored.</dd>
+	<dt><dfn>scale</dfn></dt>
+	<dd>''env()/preferred-text-scale'' returns the user's OS-level font preferences.
+	The document's initial font size incorporates both UA-level and OS-level font preferences.</dd>
+</dl>
+
+<h3 id="legacy-keyword">The 'legacy' keyword</h3>
+
+When the value of the [=text-scale=] content attribute is
+[=text-scale/legacy=]
+the user agent must set the computed ''font-size/medium'' font size
+to 16px multiplied by
+the font scale factor
+the user has chosen from any settings <i>provided by the user agent</i>.
+The ''preferred-text-scale'' value must be 1 on desktop platforms.
+On mobile:
+<ul>
+	<li>if the operating system provides a text scale setting AND the UA hasn't already applied that factor to the initial font size, ''env()/preferred-text-scale'' returns the mulitplier that the user has chosen in the operating system's text scale setting.
+
+	Note: As of this writing, all combinations of Android, iOS, Gecko, WebKit, Blink satisfy this first condition.
+
+	</li>
+	<li>Otherwise ''env()/preferred-text-scale'' returns 1.</li>
+</ul>
+
+<h3 id="scale-keyword">The 'scale' keyword</h3>
+
+The <dfn for="text-scale" export><code>scale</code></dfn> property is
+recognized in the [=text-scale=] content attribute value.
+
+When the value of the [=text-scale=] content attribute is
+<a for=text-scale>scale</a>
+computed ''font-size/medium'' font size must match the user's preferred paragraph text size as determined from a combination of OS and UA preferences.
+
+The ''env()/preferred-text-scale'' value must be a number that,
+when multiplied by 16px,
+provides a <<length>> that matches that of the initial font size.
+
+Further, when the value of the [=text-scale=] content attribute is
+<a for=text-scale>scale</a>, the user agent should skip all font-sizing interventions it would otherwise perform in an attempt to automatically honor the user's preferences. E.g. text autosizing on mobile (See [[css-size-adjust#intro]]) and full-application zoom (<a href="https://github.com/w3c/csswg-drafts/blob/main/css-env-1/explainers/env-preferred-text-scale.md#windows-11">popular browsers do this on Windows)</a>.
 
+Note: It is expected that authors will use
+''&lt;meta name="text-scale" content="scale"&gt;'' in stylesheets so that the initial font size will reflect a combination of the user's font preferences, whether those are specified at the OS level or the UA level. The author will then be able to use ''rem'' throughout the page to honor the user's font preferences.
```

Comment:
Maybe the sentence about using `rem` is actually better explained in an example block.

So you could keep the Note block to explain how authors are expected to use `scale` in their documents, with the example block below it to explain how, if you set `scale` and don't alter the default font size, content sized with `rem` units will be relative to the preferred text scale.

Also, I'm not sure if the Note should actually be an orange Assertion box. But maybe not because assertions are normative? Where I've seen Assertion boxes in the past are places like [the css-flexbox spec, where it tells authors to use the `flex` shorthand rather than the longhands.](https://www.w3.org/TR/css-flexbox/#flex-components)

Reply:
Yeah, Advisement (you meant "advisement", not "assertion". I think :) boxes are supposed to be normative. This Note seems merely descriptive as is now, but I suppose we can strengthen the language to say that authors must use it to obey the user's font preferences. Oh, though, we kind of don't care if authors use the more difficult `env(..)` instead to honor the user's preferences. So maybe we leave it non-normative?

Reply:
> So you could keep the Note block to explain how authors are expected to use `scale` in their documents, with the example block below it to explain how, if you set `scale` and don't alter the default font size, content sized with `rem` units will be relative to the preferred text scale.

I'd like to think we included a good example in the explainer we can crib from.
... and we didn't.

Yeah, an example would be good here.

Reply:
Example added.